src/main/scala/com/ossuminc/riddl/plugins/idea/files/RiddlDocumentListener.scala

```diff
@@ -7,30 +7,64 @@ import com.ossuminc.riddl.plugins.idea.files.utils.{
   getWholeWordsSubstrings,
   highlightKeywordsOnChange
 }
+import com.ossuminc.riddl.plugins.idea.utils.highlightForErrorMessage
+import com.ossuminc.riddl.plugins.idea.utils.ManagerBasedGetterUtils.*
+import com.ossuminc.riddl.plugins.idea.utils.ParsingUtils.*
 import com.ossuminc.riddl.plugins.idea.files.RiddlTokenizer.*
 
+import java.nio.file.Path
+
 class RiddlDocumentListener extends DocumentListener {
   override def documentChanged(event: DocumentEvent): Unit = {
     val doc = event.getDocument
+
     val editors = EditorFactory.getInstance().getEditors(doc)
     if editors.nonEmpty then
-      val newText = doc.getText(
-        new TextRange(event.getOffset, event.getOffset + event.getNewLength)
-      )
-      val wholeWords = getWholeWordsSubstrings(
-        doc.getText,
-        newText,
-        event.getOffset
-      )
-      highlightKeywordsOnChange(
-        wholeWords
-          .zip(
-            RiddlTokenizer
-              .tokenize(wholeWords.map(_._1).mkString(" "))
-              .filter(!_._1.isBlank)
+      editors.find(_.getDocument == doc) match
+        case Some(editor) if doc.getText.nonEmpty =>
+          val newText = doc.getText(
+            new TextRange(event.getOffset, event.getOffset + event.getNewLength)
+          )
+          val wholeWords: Seq[(String, Int)] = getWholeWordsSubstrings(
+            doc.getText,
+            newText,
+            event.getOffset
           )
-          .map((wwTup, tokTup) => (wwTup._1, wwTup._2, tokTup._3)),
-        editors.head
-      )
+          highlightKeywordsOnChange(
+            wholeWords
+              .zip(
+                RiddlTokenizer
+                  .tokenize(wholeWords.map(_._1).mkString(" "))
+                  .filter(!_._1.isBlank)
+              )
```

Comment:
THere's a replacement for this on the head of main branch now. You can tokenized the input using the TopLevelParser. It won't parse for correctness, just tokenized so should handle misspellings and bad syntax with ease.